client/src/components/HtmlEditorField/HtmlEditorField.js

```diff
@@ -0,0 +1,46 @@
+/* global window */
+import React from 'react';
+import Script from 'react-load-script';
+import { Component as TextField } from 'components/TextField/TextField';
+import fieldHolder from 'components/FieldHolder/FieldHolder';
+
+class HtmlEditorField extends TextField {
+  constructor(props) {
+    super(props);
+    this.state = {
+        isReady: false
+    };
+    this.handleReady = this.handleReady.bind(this);
+  }
+
+  getInputProps() {
+    return {
+      ...super.getInputProps(),
+      ...this.props.data.config,
+    };
+  }
+
+  handleReady() {
+    if (!window.TinyMCE && window.tinymce) {
+      window.TinyMCE = window.tinymce;
+    }
+    this.setState({ isReady: true });
```

Comment:
I recommend having `handleReady()` called in `componentWillMount()` or even set state ready in the constructor with a check

Reply:
This is contrary to the intention. `handleReady` is called only after the dependency script has been loaded into the browser (i.e. finished loading, not just rendered).

As Robbie suggested, I'll add a docblock to make this clearer.

Reply:
That sounds like an anti-pattern because the script should only load once, which is fine if you only have one instance of this component rendered.
It shouldn't need to call the script thereafter which is where the `componentWillMount()`/constructor should come in.

Reply:
I checked, the script is only loaded once. The local state ensures this is checked, but the operation I feel is safe.